clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizerService.java

```diff
@@ -0,0 +1,155 @@
+package com.netflix.spinnaker.clouddriver.deploy;
+
+import static java.lang.String.format;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.clouddriver.security.config.SecurityConfig;
+import com.netflix.spinnaker.clouddriver.security.resources.AccountNameable;
+import com.netflix.spinnaker.clouddriver.security.resources.ApplicationNameable;
+import com.netflix.spinnaker.clouddriver.security.resources.ResourcesNameable;
+import com.netflix.spinnaker.fiat.model.resources.ResourceType;
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.Errors;
+
+public class DescriptionAuthorizerService {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final Registry registry;
+  private final FiatPermissionEvaluator fiatPermissionEvaluator;
+  private final SecurityConfig.OperationsSecurityConfigurationProperties opsSecurityConfigProps;
+
+  private final Id skipAuthorizationId;
+  private final Id missingApplicationId;
+  private final Id authorizationId;
+
+  public DescriptionAuthorizerService(
+      Registry registry,
+      Optional<FiatPermissionEvaluator> fiatPermissionEvaluator,
+      SecurityConfig.OperationsSecurityConfigurationProperties opsSecurityConfigProps) {
+    this.registry = registry;
+    this.fiatPermissionEvaluator = fiatPermissionEvaluator.orElse(null);
+    this.opsSecurityConfigProps = opsSecurityConfigProps;
+
+    this.skipAuthorizationId = registry.createId("authorization.skipped");
+    this.missingApplicationId = registry.createId("authorization.missingApplication");
+    this.authorizationId = registry.createId("authorization");
+  }
+
+  public void authorize(Object description, Errors errors) {
+    authorize(description, errors, null);
+  }
+
+  public void authorize(Object description, Errors errors, ResourceType resourceType) {
```

Comment:
Thoughts on this taking a list of resource types to authorize rather than just a single one/null?

Reply:
Yep, dig it.